src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetDelegateForFunctionPointerTests.cs

```diff
@@ -0,0 +1,162 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.Emit;
+using System.Runtime.InteropServices.Tests.Common;
+using Xunit;
+
+namespace System.Runtime.InteropServices.Tests
+{
+    public class GetDelegateForFunctionPointerTests
+    {
+        [Fact]
+        public void GetDelegateForFunctionPointer_NonGeneric_ReturnsExpected()
+        {
+            MethodInfo targetMethod = typeof(GetDelegateForFunctionPointerTests).GetMethod(nameof(Method));
+            Delegate d = targetMethod.CreateDelegate(typeof(NonGenericDelegate));
+            IntPtr ptr = Marshal.GetFunctionPointerForDelegate(d);
+
+            Delegate functionDelegate = Marshal.GetDelegateForFunctionPointer(ptr, typeof(NonGenericDelegate));
+            Assert.NotNull(functionDelegate);
+            Assert.IsType<NonGenericDelegate>(functionDelegate);
+            Assert.Equal(targetMethod, functionDelegate.Method);
+            Assert.Null(functionDelegate.Target);
+        }
+
+        [Theory]
+        [InlineData(typeof(MulticastDelegate))]
+        [InlineData(typeof(OtherNonGenericDelegate))]
+        public void GetDelegateForFunctionPointer_InvalidType_ReturnsExpected(Type type)
```

Comment:
any reason for not combining this test with the above test by adding ```InlineData(typeof(NonGenericDelegate))```

Reply:
Fixed